enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,949 @@
+
+- Why do we need the kubebuilder validation tags on the Route CRD in openshift/api in addition to validation in library-go?
+    Answer:
+    The library-go validation is used for server-side validation whereas the OpenAPI schema on the Route CRD may be used for client-side validation.
+    On OCP, kube-apiserver delegates the Route API to openshift-apiserver, which employs the library-go validation to validate Route objects (see <https://github.com/openshift/openshift-apiserver/blob/aac3dd5bf0547e928103a0f718ca104b1bb13930/pkg/route/apis/route/validation/validation.go>).
+    MicroShift does not use openshift-apiserver; instead, on MicroShift, kube-apiserver handles the Route API using the Route CRD.  However, kube-apiserver has an admission plugin for the Route CRD that employs the same library-go validation that openshift-apiserver uses (see <https://github.com/openshift/kubernetes/blob/15f19ea2dd700767e5337502aec753d2a6e26905/openshift-kube-apiserver/admission/customresourcevalidation/route/validate_route.go>).
+    Defining validation for the same API in both library-go and the OpenAPI schema introduces the risk that someone could inadvertently introduce discrepancies between the library-go validation and the OpenAPI schema, leading to buggy and potentially confusing behavior.  Care must be taken to avoid introducing such discrepancies.
+    Historical note: At one time, MicroShift relied on the OpenAPI schema for server-side validation of the Route CRD, and for this reason, kubebuilder validation tags were added to the
+    Route CRD (see <https://github.com/openshift/api/pull/1228>).  Since then, an admission plugin has been added to kube-apiserver that validates the Route CRD using the same
+    library-go validation on MicroShift that openshift-apiserver uses on OpenShift (see <https://github.com/openshift/kubernetes/pull/1416>).  As a result, the kubebuilder validation tags
+    on the Route CRD are no longer required specifically for MicroShift; however, the kubebuilder validation tags are still required for client-side validation.
+
+- Why don't you set/delete the headers on frontend public_ssl in the haproxy.config?
+    Answer:
+    The connection is still encrypted in public_ssl, so it is not possible to examine or modify the HTTP session there. The fe_sni and fe_no_sni frontends terminate TLS.
+
+- Can one set load balancing algorithm using headers ?
+    Answer:
+    The HAproxy configuration provides a [load balancing algorithm](http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#4.2-balance) which relies on the
+    check against the incoming request's header values. Even though this enhancement adds the header manipulation capability it
+    doesn't aim at providing (or enhancing) the load balancing API for the IngressController or Route.  Therefore, such a
+    load balancing algorithm is out of scope for this enhancement and should be handled with a separate RFE.
```

Comment:
This looks like a duplicate of an earlier question: 

> Will this enhancement cover setting the load-balancing algorithm based on headers? Answer: No, this out of scope for this enhancement, so it will be considered as a non-goal. Please check point number 2 in the non-goals section for more details.

Please merge these two questions.

Reply:
done